client/src/components/CarouselDots/CarouselDots.js

```diff
@@ -0,0 +1,124 @@
+
+import React, {Component} from 'react';
+import "./CarouselDots.css";
+import ReactDOM from "react-dom";
+
+// Class Based React Component
+class CarouselDots extends Component{
+  constructor(props){
+    super(props);
+    console.log(props);
+
+    // Default CSS class to apply to the Component
+    this.state = {
+      classList: "CarouselDots",
+      numberOfDots: props.numberOfDots,
+      activeLocation: props.activeLocation,
+      activeDotColor: props.activeDotColor,
+      inactiveDotColor: props.inactiveDotColor,
+      hasBorder: props.hasBorder,
+      borderWeight: props.borderWeight,
+      height: 50,
+      width: 50
+
+    };
+  }
+
+
+  // Runs after Component is loaded in the broswer
+  componentDidMount() {
+    let height = ReactDOM.findDOMNode(this).parentNode.offsetHeight;
+    this.setState({
+      height: height,
+      width: height
+    })
+  }
+  // Runs after a component has been updated
+  componentDidUpdate(props){
+    console.log('Carousel Updated');
+    //console.log(this.state.dots)
+    // console.log(props);
+    // let activeLocation = this.props.activeLocation;
+    // let previousLocation = this.state.activeLocation;
+    // let newLocation = activeLocation !== previousLocation;
+    // console.log(activeLocation, previousLocation, newLocation);
+    // if(newLocation){
+    //   this.setState({activeLocation});
+    // }
+  }
+
+
+  dotClicked=(index)=>{
+    this.setState({activeLocation:index},()=>{
+      this.props.onStatusChange('Updated');
+    })
+  }
+
+  // Runs right before a component is removed from the DOM
+  componentWillUnmount(){}
+
+  generateDots=()=>{
+    let numberOfDots = this.state.numberOfDots;
+    let activeDot = this.state.activeLocation;
+    let dotArray = [];
+    let height = this.state.height;
+    let width = this.state.width;
+    let activeFill = this.state.activeDotColor;
+    let inactiveFill = this.state.inactiveDotColor;
+    let hasBorder = this.state.hasBorder;
+    let borderWeight = this.state.borderWeight;
```

Comment:
This is a lot of stuff coming from `state`/`props` - can we clean this up with destructing?

```js
let { a, b, c } = letters
```